### ISSUE TITLE
train_shadownet_multi_gpu()

### ISSUE THREAD

**Issue by soldierofhell** [MaybeShewill-CV/CRNN_Tensorflow#259]
Hi, I'm new to multi gpu trainig, so forgive me if these are basic questions. My main source of reference is classic cifar10_multi_gpu_train.py example.

1. Usually gradient aggregation is done on cpu (like in cifar10 example). In train_shadownet_multi_gpu() I don't see any "with tf.device('/cpu:0'):", so I assume it is done on (default) /gpu:1. Am I right? Why is that and what are the reasons? Maybe it just doesn't make a difference?

2. Batch normalization: here I don't understand why not to try utilize all of the towers? Maybe it's because batch_size is big enough? But let say we have a bigger net with batch_size restricted to 1 or 2 for one gpu. Then using just one gpu would be a waste of information and in general would lead to suboptimal results?

3. I still see people mostly using "manual" tower loops for multi gpu graph construction. Why not to use tf.distributed module? Is it matter of habit or tf.distributed still lacks some functionality?

Thanks in advance

**Comment by MaybeShewill-CV** [comment]
@soldierofhell 1.The aggregation op was not on cpu you may put it on cpu if you like. 2.I have not implement a bn op for distributed training process so I only use the gradient on single tower to update the params 3.I will try what you said later:)

**Comment by soldierofhell** [comment]
Thanks, I guess I'll have to experiment to compare performance (what I finally buy RTX 2070).

Nice trick in your code to set reusable flag inside compute_net_gradients() instead of setting "global" tf.get_variable_scope().reuse_variables()

**Comment by MaybeShewill-CV** [comment]
@soldierofhell I only have 1070 here==!. You may get a more fast converge process when using RTX 2070:)
